### PR TITLE
Feature/270 expanding dftool ve functionality

### PR DESCRIPTION
**Pull request by idiom-bytes**
Adding functionality to more easily interact w/ ve functions
Fixes https://github.com/oceanprotocol/df-py/issues/270

### CODE REVIEW

**Comment by idiom-bytes** [comment]
Will add tests and make sure they are passing as part of improving dftool functionality to simplify various interactions.
Changing this to draft until this is in place.

**Comment by trentmc** [comment]
This PR needs a github issue that states the problem.

(Every PR should have a related github issue. And where possible it should be 1:1. Simpler.)

File: README-dist-ops.md [hunk 1]
```diff
@@ -90,24 +90,24 @@ dftool calc $CSV_PATH 10000 OCEAN
 
 ### Step 4: Run dispense
 
-Get a working account. We call it dftool_acct. Either use a previous one, or create a new one. For the latter::
+Get a working account. We call it DFTOOL_ACCT. Either use a previous one, or create a new one. For the latter::
```
Comment (trentmc) [inline README-dist-ops.md:93]: Please move all changes wrt `dftool_acct` to a new PR. (With an appropriate github issue attached.)

Why: each PR should be well-defined and tightly scoped. Way easier to review and approve. Way easier to maintain. To be specific here: small refactorings, especially ones that have nothing to do with the main problem the PR is trying to solve, should not be in the PR.

File: util/test/test_dftool.py [hunk 2]
```diff
@@ -128,13 +128,16 @@ def test_noarg_commands():
         "dispense_active",
         "querymany",
         "compile",
-        "manyrandom",
-        "newdfrewards",
+        "many_random",
+        "new_dfrewards",
```
Comment (trentmc) [inline util/test/test_dftool.py:132]: Please move all changes wrt manyrandom, newdfrewards, newacct, newtoken, acctinfo, chaininfo to a new PR. (With an appropriate github issue attached.)

Why: each PR should be well-defined and tightly scoped. (As discussed above.)

File: dftool [hunk 3]
```diff
@@ -39,38 +44,58 @@ CHAINID_EXAMPLES = (
...
 
+# ve constants
+WEEK = 7 * 86400
+MAXTIME = 4 * 365 * 86400  # 4 years
+
 # ========================================================================
 HELP_MAIN = """Data Farming tool, for use by OPF.
 
-Usage: dftool getrate|volsym|.. ARG1 ARG2 ..
+Usage: dftool getrate|query|calc|dispense|..
```
Comment (trentmc) [inline dftool:54]: This change is incorrect. Suggest to revert. Eg. We _should_ have volsym, and _shouldn't_ have query.

File: dftool [hunk 4]
```diff
@@ -3,7 +3,7 @@
 import datetime
 import os
 import sys
-import functools
```
Comment (trentmc) [inline dftool:6]: There are a _ton_ of changes in dftool that have nothing to do with the main goal of this PR. It's super-hard for me to separate as I review.

Therefore as above: please move changes that are outside the scope of this PR's goal to >=1 new PRs. 

When you're done, please request a new review from me for this PR (and the other PRs).

**Review by trentmc — CHANGES_REQUESTED** [review]

**Comment by idiom-bytes** [comment]
It does, it's issue https://github.com/oceanprotocol/df-py/issues/270

File: dftool (hunk 3 above)
Comment (idiom-bytes) [reply dftool:54]: I see what you are referring to.

File: README-dist-ops.md (hunk 1 above)
Comment (idiom-bytes) [reply README-dist-ops.md:93]: I misread this, it looks good at is.

File: util/test/test_dftool.py (hunk 2 above)
Comment (idiom-bytes) [reply util/test/test_dftool.py:132]: Moved to this PR => https://github.com/oceanprotocol/df-py/pull/506
Ticket => https://github.com/oceanprotocol/df-py/issues/505

File: dftool (hunk 4 above)
Comment (idiom-bytes) [reply dftool:6]: Done.

**Comment by idiom-bytes** [comment]
Resolved all issues and moved everything to other PR's.
I'd like to close this PR and delete any branches associated with it.

File: dftool (hunk 4 above)
Comment (idiom-bytes) [reply dftool:6]: Ok, I see what happened... 

I'm gonna open a separate PR `issue270-new-dftool-commands` and move these commands there.
We can then close this PR/branch so nothing dirty makes it in.

**Comment by jamiehewitt15** [comment]
Can you pull from main and resolve the conflicts?

**Comment by idiom-bytes** [comment]
```Can you pull from main and resolve the conflicts?```
This PR is not supposed to be merged. Closing/deleting.
